File: 02_algorithm/programmers/Level1/(2019-KAKAO-겨울인턴십)크레인 인형뽑기 게임.py
def solution(board, moves):
    answer = 0
    array_size = len(board[0])
    stack = []
    for move in moves:
        for idx in range(array_size):
            if board[idx][move - 1] != 0:
                stack.append(board[idx][move - 1])
                board[idx][move - 1] = 0
                if len(stack) >= 2:
                    if stack[-1] == stack[-2]:
                        stack.pop()
                        stack.pop()
                        answer += 2
                break
    return answer


print(solution([[0,0,0,0,0],[0,0,1,0,3],[0,2,5,0,1],[4,2,4,4,2],[3,5,1,3,1]], [1,5,3,5,1,2,1,4]))

File: 06_js/04_algorithm/programmers/Level1/(2019-KAKAO-겨울인턴십)크레인 인형뽑기 게임.js
function solution(board, moves) {
    let answer = 0;
    let arraySize = board[0].length;
    let stack = [];
    for (let i = 0; i < moves.length; i++) {
        for (let j = 0; j < arraySize; j++) {
            if (board[j][moves[i] - 1] !== 0) {
                stack.push(board[j][moves[i] - 1]);
                board[j][moves[i] - 1] = 0;
                if (stack.length >= 2) {
                    let stackLen = stack.length;
                    if (stack[stackLen - 1] === stack[stackLen - 2]) {
                        stack.pop();
                        stack.pop();
                        answer += 2;
                    }
                }
                break;
            }
        }
    }
    return answer;
}

console.log(solution([
    [0, 0, 0, 0, 0],
    [0, 0, 1, 0, 3],
    [0, 2, 5, 0, 1],
    [4, 2, 4, 4, 2],
    [3, 5, 1, 3, 1]
], [1, 5, 3, 5, 1, 2, 1, 4]));